### ISSUE TITLE
type checking not performed at all?

### ISSUE THREAD

**Issue by iwinux** [AtomLinter/linter-tslint#192]
## Setup

* TypeScript 2.4.2
* TSLint 5.7.0
* Atom 1.19.3
* linter-tslint 1.7.0

```json
// tslint.json
{
  "extends": [
    "tslint:latest"
  ],
  "rules": {
    "no-console": false
  }
}
```

```cson
// Atom config
"*":
  "linter-tslint":
    enableSemanticRules: true
    fixOnSave: true
```

## Problem

The linter should perform type checking for `.ts` files. Sample:

```typescript
const add = (x: number, y: number): number => {
  if (x > 1) {
    return 0;
  }

  return x + null;
};

console.log(add({}, 2));
```

When running TSLint in CLI (`yarn run -- tslint --project . --type-check`), it reports type errors as expected:

```
Error at smoke.ts:1:25: 'y' is declared but never used.
Error at smoke.ts:6:14: Object is possibly 'null'.
Error at smoke.ts:9:17: Argument of type '{}' is not assignable to parameter of type 'number'.
```

But in Atom, there're no errors reported at all.

**Comment by Arcanemagus** [comment]
@iwinux Can you try updating to v1.8.1 and checking this again?
